app/waf/src/controllers/application.controller.ts

```diff
@@ -38,7 +42,18 @@ export class ApplicationController {
   async create(
     @requestBody() application: Partial<Application>,
   ): Promise<Application> {
-    return await this.applicationRepository.create(application);
+    if (!application.id) {
+      application.id = uuid();
+    }
+    if (application.wafpolicyId) {
+      await this.wafpolicyRepository.findById(application.wafpolicyId);
+    }
+
+    try {
+      return await this.applicationRepository.create(application);
+    } catch (error) {
+      throw new HttpErrors.BadRequest(error.detail);
```

Comment:
I do not know this will affect HTTP error response such as 50x, 40x etc?

Reply:
yes, here `error` is 500, but we want to to be 400 because of user's invalid input.

Reply:
it is possible not invalid input error in the future. we probably need to catch a more specific exception in the future.

Reply:
yes